visidata/errors.py

```diff
@@ -3,6 +3,7 @@
 from visidata import vd, VisiData
 
 vd.option('debug', False, 'exit on error and display stacktrace')
+vd.option('debug_stacktrace_full', False, 'for exceptions, show callstack levels above the catch')
```

Comment:
Do we need an option for this?  It seems like it would be fine to include the extra levels always.

Reply:
In my own use, I'd want the extra levels always. And I'd rather not create another unnecessary option.

My concerns would be, will long stack traces be annoying or intimidating for newer users? And will they find them difficult to copy/paste for bug reports?

Reply:
I like the top format better, without the `^^^^`.  Can they both be in the same format?  Also can we have the more relevant stacktrace first?  It's usually the inner one, which is the second one in the example you showed.

I don't mind if these errors messages are annoying for newer users; this is one of the reasons that error sheets are bound to Ctrl+E and not Shift+E (Ctrl being more "system" focused).  But I agree that longer stack traces are less ideal for copy/pasting into bug reports.  As long as it's only 2-3x as large I hope it's not too onerous.  Maybe we could use  `options.disp_expert > 1` or something?  In any event let's omit the extra option.

Reply:
Yes, we can match the format to get rid of the carets.

Since stacktraces sometimes show up on stdout of visidata, I would worry about that output being overwhelming. 
 And a bit confusing, because if users copy the text there, they will likely not expect the unusual stack trace ordering.

What if we just print one long stacktrace (the full output of `format_stack()`) for all exceptions? And handle one case specially: the most common case, where errors occur in `mainloop.py`? We can get rid of the line `ret = vd.mainloop(scr)` and everything above it. It's quite a dirty solution, but it would get rid of 16 lines of stack trace.

Reply:
To demonstrate, I've implemented these features in the commit I just pushed:  caret stripping, converting the two separate stack traces to one long one, and trimming lines above `vd.mainloop()`.

I worry about long tracebacks in particular because they're not always tucked away inside error sheets, sometimes the user sees them on stdout at exit.